integration/fuse/src/main/java/alluxio/fuse/AlluxioJniFuseFileSystem.java

```diff
@@ -640,21 +647,49 @@ private int statfsInternal(String path, Statvfs stbuf) {
     if (res != 0) {
       return res;
     }
+
+    if (mConf.getBoolean(PropertyKey.FUSE_DF_MOUNT_POINT_ENABLED)) {
+      return setFromMountInfo(path, stbuf, uri);
+    }
+
     BlockMasterInfo info = mFsStatCache.get();
     if (info == null) {
       LOG.error("Failed to statfs {}: cannot get block master info", path);
       return -ErrorCodes.EIO();
     }
+
+    setSize(stbuf, info.getCapacityBytes(), info.getFreeBytes());
+    return 0;
+  }
+
+  private int setFromMountInfo(String path, Statvfs stbuf, AlluxioURI uri) {
+    Map<String, MountPointInfo> mountPointInfoMap = mMountPointInfoCache.get();
+    if (mountPointInfoMap == null) {
+      LOG.error("Failed to statfs {}: cannot get mount point info map", path);
+      return -ErrorCodes.EIO();
+    }
+
+    MountPointInfo mountPointInfo = mountPointInfoMap.get(uri.getPath());
```

Comment:
For the Alluxio fuse scenario, we usually first alluxio fuse mount, which will mount a new path to the UFS on Alluxio. It is new for both the local filesystem and the alluxio mount point, so uri. getPath () must be the Alluxio mount point, and it does not involve the embedded mount problem.

Reply:
For hadoop 2 and 3, if HDFS is not DistributedFileSystem, does it have those APIs (e.g. hdfs.getStatus().getCapacity();)?----- Hadoop2 has these interfaces, the original use this interface and we use the version of hadoop2. As for hadoop3 version, I checked that there are also these interfaces:
https://github.com/apache/hadoop/blob/release-3.3.5-RC0/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/FileSystem.java
https://github.com/apache/hadoop/blob/release-3.3.5-RC0/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/FsStatus.java

Ozone has, but are there any other filesystem that is not Ozone and is not DistributedFileSystem does not have those APIs?-----Because this interface is a hadoop interface, as long as they are compatible with hadoop, they must have this interface.

Reply:
@flaming-archer 
```
MountPointInfo mountPointInfo = mountPointInfoMap.get(uri.getPath());
```
Looks like this conversation has extra info coming in, but will focus on the mount point info one.

I didn't quite get the meaning of your reply.

mountPointInfoMap.get(uri.getPath()); uri.getPath() may not be a mountpoint in the actual use cases. And we need to find it's closest parent to see if which of the closest ancestors is a mount point.

Reply:
I change the branch master, actually, there will be no mount point situation. So , I add a new commit, pls help to review again @LuQQiu